### PR TITLE
Update Win32Service doc for 1.0.0 version

### PR DESCRIPTION
**Pull request by macintoshplus**
This PR update the documentation for the new Win32Service version 1.0.0.

This version will be released for PHP 8.0

### CODE REVIEW

File: reference/win32service/book.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- $Revision$ -->
+<!-- $Revision: 330340 $ -->
```
Comment (Girgias) [inline reference/win32service/book.xml:2]: Seems like a revision ref sneaked into the pages

File: reference/win32service/constants.xml [hunk 2]
```diff
@@ -493,6 +493,11 @@
  </section>
 
  <section xml:id="win32service.constants.errors">
+    <note>
+    <para>
+      These constants is no longer used on Win32Service 1.0.0 and newer.
```
Comment (Girgias) [inline reference/win32service/constants.xml:498]:
```suggestion
      These constants are no longer used as of Win32Service 1.0.0.
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 3]
```diff
@@ -30,6 +30,9 @@
       <para>
        The short name of the service.
       </para>
+      <para>
+        Since the version 1.0.0, a <literal>ValueError</literal> is throw if the value is empty.
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:34]:
```suggestion
        As of version 1.0.0, a <classname>ValueError</classname> is throw if the value is empty.
```
Arguably this needs the ValueError page added before this works, but that should be done ASAP anyway.

File: reference/win32service/functions/win32-continue-service.xml [hunk 4]
```diff
@@ -49,6 +52,9 @@
   <para>
    &win32service.noerror.false.error;
   </para>
+  <para>
+    Since version 1.0.0, returns <type>void</type>.
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:56]:
```suggestion
    As of version 1.0.0, returns <type>void</type>.
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 5]
```diff
@@ -65,7 +71,11 @@
      </thead>
      <tbody>
       <row>
-       <entry>PECL win32service 0.3.0</entry>
+        <entry>1.0.0</entry>
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:74]: I think to not confuse with PHP version changelogs the PECL package name still needs to be present, @cmb69 can you confirm?

File: reference/win32service/functions/win32-delete-service.xml [hunk 6]
```diff
@@ -77,6 +83,28 @@ win32_delete_service('dummyphp');
...
+     <thead>
+      <row>
+       <entry>&Version;</entry>
+       <entry>&Description;</entry>
+      </row>
+     </thead>
+     <tbody>
+      <row>
+        <entry>1.0.0</entry>
+        <entry>Throw <literal>ValueError</literal> on wrong data in parameters. Throw <literal>Win32ServiceException</literal> on error. The function return <type>void</type>.</entry>
```
Comment (Girgias) [inline reference/win32service/functions/win32-delete-service.xml:100]:
```suggestion
        <entry>
         Throws a <classname>ValueError</classname> on invalid data in parameters.
         Throws a <classname>Win32ServiceException</classname> on error.
         The function's return type is now <type>void</type>.
        </entry>
```
Splitting into multiple lines to be under 80 chars

File: reference/win32service/functions/win32-get-last-control-message.xml [hunk 7]
```diff
@@ -43,10 +43,19 @@
  <refsect1 role="errors"><!-- {{{ -->
   &reftitle.errors;
   <para>
-   If SAPI is not <literal>"cli"</literal>, this function emits an <constant>E_ERROR</constant> level error.
+   Before version 1.0.0, if SAPI is not <literal>"cli"</literal>, this function emits an <constant>E_ERROR</constant> level error.
```
Comment (Girgias) [inline reference/win32service/functions/win32-get-last-control-message.xml:46]:
```suggestion
   Prior to version 1.0.0, if SAPI is not <literal>"cli"</literal>, this function emits an <constant>E_ERROR</constant> level error.
```
Nit, but I think the usual should be "Prior to" might misremember tho

File: reference/win32service/functions/win32-get-last-control-message.xml [hunk 8]
```diff
@@ -43,10 +43,19 @@
...
   </para>
  </refsect1><!-- }}} -->
  
+  <refsect1 role="exceptions">
+  <title>Exceptions</title>
+  <warning>
+   <para>
+    Since version 1.0.0, shall throw <type>Win32ServiceException</type> if SAPI is not <literal>"cli"</literal>
```
Comment (Girgias) [inline reference/win32service/functions/win32-get-last-control-message.xml:54]:
```suggestion
    As of version 1.0.0, will throw a <classname>Win32ServiceException</classname> if SAPI is not <literal>"cli"</literal>
```

File: reference/win32service/setup.xml [hunk 9]
```diff
@@ -8,8 +8,7 @@
  <section xml:id="win32service.requirements">
   &reftitle.required;
   <para>
-   Windows NT, Windows 2000, Windows XP or Windows Server 2003.  Any version
-   of Windows derived from Windows NT should be compatible.
+   The Windows version supported is the same as the Visual C++ redistributable package used to build PHP.
```
Comment (Girgias) [inline reference/win32service/setup.xml:11]:
```suggestion
   The versions of Windows supported are the same as the Visual C++ redistributable package used to build PHP.
```

File: reference/win32service/win32serviceexception.xml [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+ 
+ <partintro>
+ 
+<!-- {{{ win32serviceexception intro -->
+  <section xml:id="win32serviceexception.intro">
+   &reftitle.intro;
+   <para>
+    Exception thrown when an error is returned by Windows. The exception code 
+    contains the value of the older error constant. The exception message is based 
+    to the older constant name.
```
Comment (Girgias) [inline reference/win32service/win32serviceexception.xml:20]: Some grammar issues and I'm not sure what is meant, something like:
```
The exception code is equal to the error constant returned by Windows.
The exception message is based on the constant name.
```
?

**Review by Girgias — CHANGES_REQUESTED** [review]
Haven't annotated each individual line as some of the comments are applicable to many other instances.

File: reference/win32service/book.xml (hunk 1 above)
Comment (macintoshplus) [reply reference/win32service/book.xml]: Changed

File: reference/win32service/constants.xml (hunk 2 above)
Comment (macintoshplus) [reply reference/win32service/constants.xml:498]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 3 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 4 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml:56]: Changed

File: reference/win32service/functions/win32-delete-service.xml (hunk 6 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-delete-service.xml]: Changed

File: reference/win32service/functions/win32-get-last-control-message.xml (hunk 7 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-get-last-control-message.xml]: Changed

File: reference/win32service/functions/win32-get-last-control-message.xml (hunk 8 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-get-last-control-message.xml]: changed

File: reference/win32service/setup.xml (hunk 9 above)
Comment (macintoshplus) [reply reference/win32service/setup.xml]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 5 above)
Comment (cmb69) [reply reference/win32service/functions/win32-continue-service.xml:74]: Yes; that was very recently fixed for all packages, so should be kept. It's particularly important for the generated [changelog in the appendix](https://www.php.net/manual/en/doc.changelog.php).

File: reference/win32service/win32serviceexception.xml (hunk 10 above)
Comment (macintoshplus) [reply reference/win32service/win32serviceexception.xml]: Previously, all error code is a PHP constant like this:

```php
const ERROR_INVALID_HANDLE = 0x00000006;
```
as of version 1.0.0, this error is converted in exception created like this:

```php
new Win32ServiceException('Error invalid handle', 0x00000006);
```
It's suffisant explation to understand?

Note: sorry for my english...

File: reference/win32service/functions/win32-continue-service.xml (hunk 5 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: I have add prefix on all changelog entry.

File: reference/win32service/functions/win32-delete-service.xml (hunk 6 above)
Comment (cmb69) [reply reference/win32service/functions/win32-delete-service.xml]: Note that the 80 characters are rather a soft limit; up to 100 characters may be quite okay as well.

File: reference/win32service/functions/win32-send-custom-control.xml [hunk 11]
```diff
@@ -39,6 +42,9 @@
      <para>
       The custom contole value between 128 and 255.
      </para>
+      <para>
+        As of version 1.0.0, a <classname>ValueError</classname> is throw if the value is not between 128 and 255.
+      </para>
```
Comment (Girgias) [inline reference/win32service/functions/win32-send-custom-control.xml:47]: Now that I think about it, wouldn't it be better to move this to the  `<refsect1 role="errors">` section?

File: reference/win32service/win32serviceexception.xml (hunk 10 above)
Comment (Girgias) [reply reference/win32service/win32serviceexception.xml]: How about this?
```
The exception replaces the old mechanism where the error value needed to be compared to constants to detect which error was emitted. The Exception code is equal to the value of the error value and the exception message is based on the corresponding constant name.
```

Comment (macintoshplus) [reply reference/win32service/win32serviceexception.xml]: Yes

File: reference/win32service/functions/win32-send-custom-control.xml (hunk 11 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-send-custom-control.xml:47]: @Girgias I have moved this error into `errors` section. It's good ?

File: reference/win32service/constants.xml [hunk 12]
```diff
@@ -493,6 +493,11 @@
  </section>
 
  <section xml:id="win32service.constants.errors">
+    <note>
+    <para>
+      These constants are no longer used as of Win32Service 1.0.0.
```
Comment (Girgias) [inline reference/win32service/constants.xml:498]:
```suggestion
      These constants are no longer in use as of Win32Service 1.0.0.
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 13]
```diff
@@ -43,15 +43,32 @@
...
- 
+
  <refsect1 role="returnvalues">
   &reftitle.returnvalues;
   <para>
    &win32service.noerror.false.error;
   </para>
+  <para>
+    As of version 1.0.0, returns <type>void</type>.
+  </para>
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:54]:
```suggestion
  <para>
   &return.void;
  </para>
  <para>
   Prior to version 1.0.0, &win32service.noerror.false.error;
  </para>
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 14]
```diff
@@ -64,6 +81,14 @@
...
      </thead>
      <tbody>
+      <row>
+        <entry>PECL win32service 1.0.0</entry>
+        <entry>
+          Throw <classname>ValueError</classname> on wrong data in parameters.
+          Throw <classname>Win32ServiceException</classname> on error.
+          The function return <type>void</type>.
+        </entry>
+      </row>
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:91]:
```suggestion
      <row>
       <entry>PECL win32service 1.0.0</entry>
       <entry>
        Throws a <classname>ValueError</classname> on invalid data in parameters,
        previously &false; was returned.
       </entry>
      </row>
      <row>
       <entry>PECL win32service 1.0.0</entry>
       <entry>
        Throws a <classname>Win32ServiceException</classname> on error,
        previously &false; was returned.
       </entry>
      </row>
      <row>
       <entry>PECL win32service 1.0.0</entry>
       <entry>
        The return type is now <type>void</type>, previously it was (describe type here).
       </entry>
      </row>
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 15]
```diff
@@ -43,15 +43,32 @@
...
+  </para>
  </refsect1>
- 
- 
+
+
+ <refsect1 role="errors"><!-- {{{ -->
+  &reftitle.errors;
+  <para>
+    As of version 1.0.0, a <classname>ValueError</classname> is throw if the
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:61]:
```suggestion
    A <classname>ValueError</classname> is thrown if the 
```

File: reference/win32service/functions/win32-continue-service.xml [hunk 16]
```diff
@@ -43,15 +43,32 @@
...
+ <refsect1 role="errors"><!-- {{{ -->
+  &reftitle.errors;
+  <para>
+    As of version 1.0.0, a <classname>ValueError</classname> is throw if the 
+    value of <parameter>servicename</parameter> parameter is empty.
+  </para>
+
+   <para>
+    As of version 1.0.0, will throw a 
+    <classname>Win32ServiceException</classname> on error.
```
Comment (Girgias) [inline reference/win32service/functions/win32-continue-service.xml:67]:
```suggestion
    A <classname>Win32ServiceException</classname> is thrown on error.
```

File: reference/win32service/functions/win32-create-service.xml [hunk 17]
```diff
@@ -295,8 +295,68 @@
   <para>
    &win32service.noerror.false.error;
   </para>
+  <para>
+    As of version 1.0.0, return <type>void</type>.
+  </para>
```
Comment (Girgias) [inline reference/win32service/functions/win32-create-service.xml:300]: Same as above

File: reference/win32service/functions/win32-create-service.xml [hunk 18]
```diff
@@ -295,8 +295,68 @@
...
+  <para>
+    As of version 1.0.0, a <classname>ValueError</classname> is throw if the 
+    value of <parameter>recovery_reset_period</parameter> parameter is not 
+    between 0 and PHP_INT_MAX.
+  </para>
+
+   <para>
+    As of version 1.0.0, will throw a 
+    <classname>Win32ServiceException</classname> on error.
+   </para>
```
Comment (Girgias) [inline reference/win32service/functions/win32-create-service.xml:355]: Same as above (also indentation nits)

File: reference/win32service/functions/win32-create-service.xml [hunk 19]
```diff
@@ -386,6 +446,14 @@ debug_zval_dump($x);
...
      </thead>
      <tbody>
+      <row>
+        <entry>PECL win32service 1.0.0</entry>
+        <entry>
+          Throw <classname>ValueError</classname> on wrong data in parameters.
+          Throw <classname>Win32ServiceException</classname> on error.
+          The function return <type>void</type>.
+        </entry>
+      </row>
```
Comment (Girgias) [inline reference/win32service/functions/win32-create-service.xml:456]: Same as above (split into multiple rows)

File: reference/win32service/functions/win32-set-service-exit-code.xml [hunk 20]
```diff
@@ -53,9 +53,41 @@
  <refsect1 role="errors"><!-- {{{ -->
   &reftitle.errors;
   <para>
-   If SAPI is not <literal>"cli"</literal>, this function emits an 
+   Prior to version 1.0.0
+, if SAPI is not <literal>"cli"</literal>, this function emits an
```
Comment (Girgias) [inline reference/win32service/functions/win32-set-service-exit-code.xml:57]:
```suggestion
   Prior to version 1.0.0, if the SAPI is not <literal>"cli"</literal>, this function emits an
```

File: reference/win32service/functions/win32-set-service-exit-mode.xml [hunk 21]
```diff
@@ -14,14 +14,14 @@
    <methodparam choice="opt"><type>bool</type><parameter>gracefulMode</parameter><initializer>true</initializer></methodparam>
   </methodsynopsis>
   <para>
-   If <parameter>gracefulMode</parameter> parameter is provided, the exit mode 
-is changed. When the exit mode is not gracefuly, the exit code used can be set 
+   If <parameter>gracefulMode</parameter> parameter is provided, the exit mode
+is changed. When the exit mode is not gracefuly, the exit code used can be set
```
Comment (Girgias) [inline reference/win32service/functions/win32-set-service-exit-mode.xml:18]:
```suggestion
   If <parameter>gracefulMode</parameter> parameter is provided, the exit mode
   is changed. When the exit mode is not gracefuly, the exit code used can be set
```
Nit indentation

File: reference/win32service/functions/win32-set-service-exit-mode.xml [hunk 22]
```diff
@@ -51,9 +51,43 @@ with the <function>win32_set_service_exit_code</function> function.
  <refsect1 role="errors"><!-- {{{ -->
   &reftitle.errors;
   <para>
-   If SAPI is not <literal>"cli"</literal>, this function emits an 
+   Prior to version 1.0.0
+, if SAPI is not <literal>"cli"</literal>, this function emits an
```
Comment (Girgias) [inline reference/win32service/functions/win32-set-service-exit-mode.xml:55]:
```suggestion
   Prior to version 1.0.0, if SAPI is not <literal>"cli"</literal>, this function emits an
```

File: reference/win32service/functions/win32-start-service-ctrl-dispatcher.xml [hunk 23]
```diff
@@ -77,10 +77,18 @@
  <refsect1 role="errors"><!-- {{{ -->
   &reftitle.errors;
   <para>
-   If SAPI is not <literal>"cli"</literal>, this function emits an <constant>E_ERROR</constant> level error.
+   Prior to version 1.0.0
+, if SAPI is not <literal>"cli"</literal>, this function emits an 
+<constant>E_ERROR</constant> level error.
```
Comment (Girgias) [inline reference/win32service/functions/win32-start-service-ctrl-dispatcher.xml:82]:
```suggestion
   Prior to version 1.0.0, if SAPI is not <literal>"cli"</literal>, this function emits an 
   <constant>E_ERROR</constant> level error.
```

File: reference/win32service/versions.xml [hunk 24]
```diff
@@ -17,6 +17,8 @@
  <function name='win32-start-service-ctrl-dispatcher' from='PECL win32service &gt;=0.1.0'/>
  <function name='win32-start-service' from='PECL win32service &gt;=0.1.0'/>
  <function name='win32-stop-service' from='PECL win32service &gt;=0.1.0'/>
+
```
Comment (Girgias) [inline reference/win32service/versions.xml:20]: Unnecessary line break

**Review by Girgias — CHANGES_REQUESTED** [review]
Sorry took me while to get round to this.

Please pay attention to indentation which is 1 space per level in XML, 4 spaces for PHP examples.
If you can update the sections and changelogs similarly to what I've done for the first example, as we want the most recent versions to be on top.

File: reference/win32service/functions/win32-continue-service.xml (hunk 14 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 13 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: Changed

File: reference/win32service/constants.xml (hunk 12 above)
Comment (macintoshplus) [reply reference/win32service/constants.xml]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 15 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: Changed

File: reference/win32service/functions/win32-continue-service.xml (hunk 16 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-continue-service.xml]: Changed

File: reference/win32service/functions/win32-create-service.xml (hunk 17 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-create-service.xml]: Changed

File: reference/win32service/functions/win32-create-service.xml (hunk 18 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-create-service.xml]: All changed

File: reference/win32service/functions/win32-create-service.xml (hunk 19 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-create-service.xml]: All changed

File: reference/win32service/functions/win32-set-service-exit-code.xml (hunk 20 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-set-service-exit-code.xml]: Changed

File: reference/win32service/functions/win32-set-service-exit-mode.xml (hunk 22 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-set-service-exit-mode.xml]: Changed

File: reference/win32service/functions/win32-start-service-ctrl-dispatcher.xml (hunk 23 above)
Comment (macintoshplus) [reply reference/win32service/functions/win32-start-service-ctrl-dispatcher.xml]: Changed

File: reference/win32service/versions.xml (hunk 24 above)
Comment (macintoshplus) [reply reference/win32service/versions.xml]: Changed

**Review by Girgias — APPROVED** [review]
LGTM, do you have commit karma on doc-en?

**Comment by cmb69** [comment]
Thank you for the PR! I reverted most of the unrelated whitespace changes, to make it a bit easier for translators (who certainly have a lot of work to do right now, due to PHP 8).

**Pull request closed without merging**